sw/host/tests/chip/spi_device_smoketest/src/main.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use regex::Regex;
+use std::fs;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use object::{Object, ObjectSection, ObjectSymbol};
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::spi::Transfer;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::uart::console::{ExitStatus, UartConsole};
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Name of the SPI interface to connect to the OTTF console.
+    #[arg(long, default_value = "BOOTSTRAP")]
+    spi: String,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+}
+
+fn spi_device_tx_rx_test(
+    opts: &Opts,
+    transport: &TransportWrapper,
+    tx_data: &[u8],
+    rx_data: &[u8],
+) -> Result<()> {
+    let uart = transport.uart("console")?;
+
+    /* Wait sync message. */
```

Comment:
If I follow correctly, there device code contains 5 "subtests". Is it be possible to split this function into 5 and use `execute_test!`? This would make it easier to follow which part of the host code corresponds to which part of the device code. If not, maybe add comments saying `this corresponds to the `tx_watermark_test test in device code`.

Reply:
Good point, done.